### PR TITLE
Remove lib commons

### PR DESCRIPTION
**Pull request by DreierF**
- [ ] [No new Teamscale findings](https://build.cqse.eu/teamscale/delta.html#input/jacoco-client/?showMergeFindings=true)
- [ ] [Changes are tested adequately](https://build.cqse.eu/teamscale/tests.html#/jacoco-client/)
- [ ] README.md updated in case of user-visible changes
- [ ] CHANGELOG.md updated
- [ ] build.gradle version number increased according to semantic versioning

### CODE REVIEW

File: report-generator/src/main/java/com/teamscale/report/testwise/jacoco/cache/ClassCoverageLookup.java [hunk 1]
```diff
@@ -12,11 +12,10 @@
...
  * <p>
- * - Create an instance of this class for every analyzed java class.
- * - Set the file name of the java source file from which the class has been created.
- * - Then call {@link #addProbe(int, Set)} for all probes and lines that belong to that probe.
- * - Afterwards call {@link #getFileCoverage(ExecutionData, ILogger)} to transform probes ({@link ExecutionData}) for
- * this class into covered lines ({@link FileCoverageBuilder}).
+ * - Create an instance of this class for every analyzed java class. - Set the file name of the java source file from
+ * which the class has been created. - Then call {@link #addProbe(int, Set)} for all probes and lines that belong to
+ * that probe. - Afterwards call {@link #getFileCoverage(ExecutionData, ILogger)} to transform probes ({@link
+ * ExecutionData}) for this class into covered lines ({@link FileCoverageBuilder}).
```
Comment (albertsteckermeier) [inline report-generator/src/main/java/com/teamscale/report/testwise/jacoco/cache/ClassCoverageLookup.java]: Formatting of bullets broken. Maybe needs <br/> at the end?

File: teamscale-client/src/main/java/com/teamscale/client/ImmutablePair.java [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+public class ImmutablePair<S, T> implements Cloneable, Comparable<ImmutablePair<S, T>>, Serializable {
+
+	/** Version used for serialization. */
+	private static final long serialVersionUID = 1;
+
+	/** The first element. */
+	protected S first;
+
+	/** The second element. */
+	protected T second;
```
Comment (albertsteckermeier) [inline teamscale-client/src/main/java/com/teamscale/client/ImmutablePair.java]: These fields should be final in ImmutablePair.

File: teamscale-client/src/main/java/com/teamscale/client/StringUtils.java [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+	public static final String EMPTY_STRING = "";
+
+	/** Number formatter. */
+	private static NumberFormat numberFormat = NumberFormat.getInstance();
+
+	/**
+	 * Format number
+	 */
+	public static String format(Number number) {
+		return numberFormat.format(number);
```
Comment (albertsteckermeier) [inline teamscale-client/src/main/java/com/teamscale/client/StringUtils.java]: NumberFormat is not thread safe i think

File: teamscale-gradle-plugin/src/main/kotlin/com/teamscale/BuildVersion.kt [hunk 4]
```diff
@@ -5,7 +5,7 @@ object BuildVersion {
 
     /** Extracts the plugin's version from the jar's manifest file. */
     val buildVersion: String
-        get() = "0.4.8"
+        get() = "0.4.10"
```
Comment (albertsteckermeier) [inline teamscale-gradle-plugin/src/main/kotlin/com/teamscale/BuildVersion.kt]: Is 0.4.11 in Gitlab MR.

File: teamscale-client/src/main/java/com/teamscale/client/PairList.java [hunk 5]
```diff
@@ -0,0 +1,312 @@
...
+import java.util.Map.Entry;
+import java.util.Objects;
+import java.util.function.BiConsumer;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+
+/**
+ * A list for storing pairs in a specific order.
+ */
+public class PairList<S, T> implements Serializable, Iterable<Pair<S, T>> {
```
Comment (albertsteckermeier) [inline teamscale-client/src/main/java/com/teamscale/client/PairList.java]: Do we really need the PairList? Maybe List<Pair> would be sufficient.

File: build.gradle [hunk 6]
```diff
@@ -12,9 +12,9 @@ buildscript {
 
 ext {
 	appVersion = '11.3.1' // Also update version in BuildVersion.kt
-	pluginVersion = '0.4.8' // Also update version in BuildVersion.kt
-	reportGeneratorVersion = '1.3.6'
-	teamscaleClientVersion = '0.2.8'
+	pluginVersion = '0.4.10' // Also update version in BuildVersion.kt
```
Comment (albertsteckermeier) [inline build.gradle]: Gitlab MR uses 0.4.11.

File: teamscale-client/src/main/java/com/teamscale/client/PairList.java (hunk 5 above)
Comment (DreierF) [reply teamscale-client/src/main/java/com/teamscale/client/PairList.java]: Just wanted to make the changes as minimal as possible, but you are right we only need it in one place that can as well be solved with a regular list.

**Pull request merged**